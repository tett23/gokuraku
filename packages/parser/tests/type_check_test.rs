extern crate parser;

use insta::{assert_debug_snapshot, glob};
use std::fs;

#[test]
fn type_check_test() {
    glob!("../fixtures/pds", "*.pds", |path| {
        dbg!(&path);
        let input = fs::read_to_string(path).unwrap();
        let ast = parser::prose_down_script_parse(&input).unwrap();
        let ir1 = parser::ir::ir1::transform1(&ast);
        let ir2 = parser::ir::ir2::transform2(ir1, ast);

        assert_debug_snapshot!(ir2);
    });
}
