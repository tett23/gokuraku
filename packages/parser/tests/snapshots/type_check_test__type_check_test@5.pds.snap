---
source: packages/parser/tests/type_check_test.rs
expression: ir2
input_file: packages/parser/fixtures/pds/5.pds
---
Ok(
    IR2 {
        ident_symbols: SymbolTable2(
            {
                Ident(
                    "main",
                ): (
                    [
                        Function {
                            ident: Ident(
                                "main",
                            ),
                            expr: Abstruction {
                                arg: None,
                                expr: Literal(
                                    Text(
                                        "Hello world",
                                    ),
                                ),
                            },
                            where_clause: IR2 {
                                ident_symbols: SymbolTable2(
                                    {},
                                ),
                                data_symbols: SymbolTable2(
                                    {},
                                ),
                                handler_symbols: SymbolTable2(
                                    {},
                                ),
                                trait_symbols: SymbolTable2(
                                    {},
                                ),
                                inst_symbols: SymbolTable2(
                                    {},
                                ),
                            },
                        },
                    ],
                    None,
                ),
            },
        ),
        data_symbols: SymbolTable2(
            {},
        ),
        handler_symbols: SymbolTable2(
            {},
        ),
        trait_symbols: SymbolTable2(
            {},
        ),
        inst_symbols: SymbolTable2(
            {},
        ),
    },
)
